build/config/linux/pkg_config.gni

```diff
@@ -137,10 +137,16 @@ template("pkg_config") {
 
       # Link libraries statically for OSS-Fuzz fuzzer build
       if (oss_fuzz) {
+        #outputs libs needed for static linking (direct + transitive/non-direct libs), we will re-execute the script to get those libs
+        args += [ "--static" ]
+        pkgresult_static = exec_script(pkg_config_script, args, "value")
         libs = []
         ldflags = [ "-Wl,-Bstatic" ]
-        foreach(lib, pkgresult[3]) {
-          ldflags += [ "-l$lib" ]
+        foreach(lib, pkgresult_static[3]) {
+          #dl(dynamic loading) lib is not needed for linking statically and its presence triggers errors.
```

Comment:
```suggestion
          # dl(dynamic loading) lib is not needed for linking statically and its presence triggers errors.
```

Reply:
what errors? Is there a reference link we can add for this?

Reply:
Not Really, I saw those errors locally.
Here is an example after keeping `-ldl` in the command
```
[765/811] ld tests/fuzz-setup-payload-base38-decode
FAILED: tests/fuzz-setup-payload-base38-decode 
clang++ -O0 -fPIC -Werror -Wl,--fatal-warnings -Wl,-z,defs -fdiagnostics-color -O1 -fno-omit-frame-pointer -gline-tables-only -Wno-error=enum-constexpr-conversion -Wno-error=incompatible-function-pointer-types -Wno-error=int-conversion -Wno-error=deprecated-declarations -Wno-error=implicit-function-declaration -Wno-error=implicit-int -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=memory -fsanitize-memory-track-origins -fsanitize=fuzzer-no-link -stdlib=libc++ -fsanitize=fuzzer -fuse-ld=lld -Wl,--gc-sections -pie -O1 -fno-omit-frame-pointer -gline-tables-only -Wno-error=enum-constexpr-conversion -Wno-error=incompatible-function-pointer-types -Wno-error=int-conversion -Wno-error=deprecated-declarations -Wno-error=implicit-function-declaration -Wno-error=implicit-int -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=memory -fsanitize-memory-track-origins -fsanitize=fuzzer-no-link -stdlib=libc++ -fsanitize=fuzzer -Wl,-Bstatic -lssl -lcrypto -ldl -Wl,-Bdynamic @tests/fuzz-setup-payload-base38-decode.rsp -o tests/fuzz-setup-payload-base38-decode
ld.lld: error: undefined symbol: __dlsym
>>> referenced by dlsym.o:(dlsym) in archive /lib/x86_64-linux-gnu/libdl.a

ld.lld: error: undefined symbol: __dlvsym
>>> referenced by dlvsym.o:(dlvsym) in archive /lib/x86_64-linux-gnu/libdl.a
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
I would add the text like `ld.lld: error: undefined symbol: __dlvsym` to the comment (or maybe even more line out of it) so that it is easily greppable if others hit it.